kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java

```diff
@@ -253,55 +255,48 @@ public void uploadFile() throws IOException {
 
     client.pods().inNamespace(session.getNamespace()).withName(pod1.getMetadata().getName())
-      .file("/tmp/toBeUploaded").upload(tmpFile.toPath());
-
-    try (
-      final InputStream checkIs = client.pods().inNamespace(session.getNamespace())
-        .withName(pod1.getMetadata().getName()).file("/tmp/toBeUploaded").read();
-      final ByteArrayOutputStream resultOs = new ByteArrayOutputStream()
-    ) {
-      IOUtils.copy(checkIs, resultOs);
-      assertEquals(resultOs.toString(StandardCharsets.UTF_8.name()),"I'm uploaded");
+      .file("/tmp/toBeUploaded").upload(tmpFile);
+
+    try (final InputStream checkIs = client.pods().inNamespace(session.getNamespace())
+        .withName(pod1.getMetadata().getName()).file("/tmp/toBeUploaded").read()) {
+      String result = new BufferedReader(new InputStreamReader(checkIs, StandardCharsets.UTF_8))
+        .lines().collect(Collectors.joining(System.lineSeparator()));
+      assertEquals("I'm uploaded", result);
     }
   }
 
   @Test
-  public void uploadDir() {
+  public void uploadDir() throws IOException {
     // Wait for resources to get ready
     Pod pod1 = client.pods().inNamespace(session.getNamespace()).withName("pod-standard").get();
     ReadyEntity<Pod> podReady = new ReadyEntity<>(Pod.class, client, pod1.getMetadata().getName(), session.getNamespace());
     await().atMost(POD_READY_WAIT_IN_SECONDS, TimeUnit.SECONDS).until(podReady);
 
     final String[] files = new String[]{"1", "2"};
-    final File tmpDir = Files.createTempDir();
-    final File uploadDir = new File(tmpDir, "uploadDir");
-    assertTrue(uploadDir.mkdir());
-    Stream.of(files).map(fileName -> new File(uploadDir, fileName)).forEach(f -> {
+    final Path tmpDir = Files.createTempDirectory("uploadDir");
+    Stream.of(files).map(fileName -> tmpDir.resolve(fileName)).forEach(f -> {
       try {
-        assertTrue(f.createNewFile());
-        Files.write("I'm uploaded", f, StandardCharsets.UTF_8);
+        Path file = Files.createFile(f);
+        assertTrue(Files.exists(file));
+        Files.write(file, Arrays.asList("I'm uploaded"));
       } catch (IOException ex) {
         fail(ex.getMessage());
       }
     });
 
     client.pods().inNamespace(session.getNamespace()).withName(pod1.getMetadata().getName())
-      .dir("/tmp/uploadDir").upload(uploadDir.toPath());
+      .dir("/tmp/uploadDir").upload(tmpDir);
 
     Stream.of(files).forEach(fileName -> {
-      try (
-        final InputStream checkIs = client.pods().inNamespace(session.getNamespace())
-          .withName(pod1.getMetadata().getName()).file("/tmp/uploadDir/"+fileName).read();
-        final ByteArrayOutputStream resultOs = new ByteArrayOutputStream()
-      ) {
-        IOUtils.copy(checkIs, resultOs);
-        assertEquals(resultOs.toString(StandardCharsets.UTF_8.name()),"I'm uploaded");
+      try (final InputStream checkIs = client.pods().inNamespace(session.getNamespace())
```

Comment:
Could we move the Reader instantiation to the resources block so it gets closed automatically.